presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -15,259 +15,195 @@
-    private static final String LOCAL_URI = "http://127.0.0.1/v1/task/";
 
     private final Session session;
     private final TaskId taskId;
     private final PlanFragment planFragment;
     private final OutputBuffers outputBuffers;
-    private final HttpClient httpClient;
+    private final PrestoSparkHttpWorkerClient workerClient;
     private final TableWriteInfo tableWriteInfo;
-    private final RemoteTaskStats stats;
     private final List<TaskSource> sources;
     private final Executor executor;
     private final ScheduledExecutorService updateScheduledExecutor;
-    private final JsonCodec<TaskUpdateRequest> taskUpdateRequestCodec;
-    private final JsonCodec<PlanFragment> planFragmentCodec;
-    private final JsonCodec<TaskInfo> taskInfoCodec;
-    private final AtomicReference<TaskInfo> taskInfo = new AtomicReference<>();
-
-    @GuardedBy("this")
-    private ScheduledFuture<?> scheduledFuture;
+    private final HttpNativeExecutionTaskInfoFetcher taskInfoFetcher;
+    private final HttpNativeExecutionTaskResultFetcher taskResultFetcher;
 
     public NativeExecutionTask(
             Session session,
+            URI location,
             TaskId taskId,
             PlanFragment planFragment,
             List<TaskSource> sources,
             HttpClient httpClient,
             TableWriteInfo tableWriteInfo,
-            RemoteTaskStats stats,
             Executor executor,
-            ScheduledExecutorService updateScheduledExecutor,
-            JsonCodec<TaskUpdateRequest> taskUpdateRequestCodec,
-            JsonCodec<PlanFragment> planFragmentCodec,
-            JsonCodec<TaskInfo> taskInfoCodec)
+            ScheduledExecutorService updateScheduledExecutor)
     {
         requireNonNull(session, "session is null");
         requireNonNull(taskId, "taskId is null");
         requireNonNull(planFragment, "planFragment is null");
         requireNonNull(sources, "sources is null");
         requireNonNull(httpClient, "httpClient is null");
         requireNonNull(tableWriteInfo, "tableWriteInfo is null");
-        requireNonNull(stats, "stats is null");
         requireNonNull(executor, "executor is null");
         requireNonNull(updateScheduledExecutor, "updateScheduledExecutor is null");
-        requireNonNull(taskUpdateRequestCodec, "taskUpdateRequestCodec is null");
-        requireNonNull(planFragmentCodec, "planFragmentCodec is null");
-        requireNonNull(taskInfoCodec, "taskInfoCodec is null");
 
         try (SetThreadName ignored = new SetThreadName("NativeExecutionTask-%s", taskId)) {
             this.taskId = taskId;
             this.session = session;
             this.planFragment = planFragment;
-            this.httpClient = httpClient;
             this.tableWriteInfo = tableWriteInfo;
-            this.stats = stats;
             this.sources = sources;
             this.executor = executor;
             this.updateScheduledExecutor = updateScheduledExecutor;
```

Comment:
This is the one I'm talking about regarding the above comment. You don't need the assignment here. Just pass `updateScheduledExecutor` directly into the below constructors.